x-pack/plugins/security_solution/server/endpoint/routes/metadata/handlers.ts

```diff
@@ -85,6 +86,10 @@ export const getMetadataRequestHandler = function (
         ),
       });
     } catch (error) {
+      if (error instanceof EndpointHostNotFoundError) {
+        logger?.debug(`agent with id ${request.params.id} not found`);
+        return response.notFound({ body: error });
+      }
```

Comment:
Wondering if we should move this error to the `errorHandler()` instead, since that is meant to handle all errors 🤔

Reply:
Yeah I too think we should move this specific error instance block to `errorHandler`.

Reply:
@paul-tavares @ashokaditya - I moved the error into `errorHandler()`.  I also adjusted `errorHandler()` to not always log to `error` since we don't want to log to error on this.